File: tut-vaadin-simple-navigator/src/main/java/com/github/galcyurio/nav/ui/MainUI.java
package com.github.galcyurio.nav.ui;

import com.github.galcyurio.nav.ui.home.HomeView;
import com.github.galcyurio.nav.ui.start.StartView;
import com.vaadin.navigator.Navigator;
import com.vaadin.server.VaadinRequest;
import com.vaadin.spring.annotation.SpringUI;
import com.vaadin.ui.UI;

/**
 * @author galcyurio
 */
@SpringUI
public class MainUI extends UI {
    Navigator navigator;

    @Override
    protected void init(VaadinRequest request) {
        getPage().setTitle("Navigation Example");

        // Create a navigator to control the views
        navigator = new Navigator(this, this);

        // Create and register the views
        navigator.addView(StartView.VIEW_NAME, new StartView());
        navigator.addView(HomeView.VIEW_NAME, new HomeView());
    }


}

File: tut-vaadin-simple-push-state-navigation/src/main/kotlin/com/github/galcyurio/ui/MainUI.kt
package com.github.galcyurio.ui

import com.vaadin.navigator.Navigator
import com.vaadin.navigator.PushStateNavigation
import com.vaadin.server.VaadinRequest
import com.vaadin.spring.annotation.SpringUI
import com.vaadin.ui.*

/**
 * @author galcyurio
 */
@PushStateNavigation
@SpringUI
class MainUI : UI() {
    override fun init(p0: VaadinRequest?) {
        val container = CssLayout()
        val navigator = Navigator(this, container)
        navigator.addView(HomeScreen.VIEW_NAME, HomeScreen::class.java)
        navigator.addView(FooScreen.VIEW_NAME, FooScreen::class.java)
        navigator.addView(BarScreen.VIEW_NAME, BarScreen::class.java)

        content = VerticalLayout().apply {
            addComponent(HorizontalLayout().apply {
                addComponent(Button("Home", Button.ClickListener { navigator.navigateTo(HomeScreen.VIEW_NAME) }))
                addComponent(Button("Foo", Button.ClickListener { navigator.navigateTo(FooScreen.VIEW_NAME) }))
                addComponent(Button("Bar", Button.ClickListener { navigator.navigateTo(BarScreen.VIEW_NAME) }))
            })
            addComponent(Label("Try refresh (F5)"))
            addComponent(container)
        }
    }
}
